src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
+
+type Color = 'red' | 'green' | 'blue';
+type Snape = 'triangle'| 'circle'| 'rectangle';
+
 export interface Figure {
+  shape: Snape,
+  color: Color,
 
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Snape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sidesSum = this.a + this.b + this.c;
+    const maxSide = Math.max(this.a, this.b, this.c);
+    const sumTwoSmallerSides = sidesSum - maxSide;
+
+    if (a <= 0 || b <= 0 || c <= 0 || maxSide >= sumTwoSmallerSides) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
 
+  public getArea(): number {
+    const sidesSum = this.a + this.b + this.c;
+
+    return Math.floor(Math.sqrt(sidesSum / 2
+      * (sidesSum / 2 - this.a)
+      * (sidesSum / 2 - this.b)
+      * (sidesSum / 2 - this.c)) * 100) / 100;
```

Comment:
It would be easier to calculate the semi-perimeter and use it instead of dividing the perimeter by two every time.